features/dynamic-bundle-discounts.md

```diff
@@ -0,0 +1,195 @@
+# Dynamic Bundle discounts
+
+- [Dynamic Bundle discounts](#dynamic-bundle-discounts)
+  - [Introduction](#introduction)
+  - [Setup](#setup)
+    - [Dynamic Bundle template changes](#dynamic-bundle-template-changes)
+    - [Shopify line item script changes](#shopify-line-item-script-changes)
+    - [Known Issues](#known-issues)
+
+## Introduction
+This documentation will explain the process of settings up `Nosto - Dynamic Bundles` for automatic discount, when a bundle is added to the cart on a Shopify store. As per the existing setup, this has to be done manually from `setting up line item script` to `script for including discount details to shopify cart API`. As a first step into completely automating this process end-to-end, Nosto has released some enhancements to internal processing of dynamic bundles that will help automate this process with minimal manual intervention.
+
+## Setup
+
+### Dynamic Bundle template changes
+A new Nosto JavaScript API `addBundleToCartWithDiscount` has been introduced that will collect the bundle information, from the bundle template, **validates and prepares the data** and then invokes the Shopify API. In order to use this new API, invoke the API from the dynamic bundle template with the following
+1. Products selected in the bundle while adding to cart
+2. Discount configuration
+3. All the product IDs from the bundle (both selected and unselected)
+
+An example where all the products from a bundle are added to cart.
+
+```javascript
+_targetWindow.Nosto.addBundleToCartWithDiscount({
+    items: [{
+        productId: '123456789',
+        quantity: 1
+    }, {
+        productId: '987654321',
+        quantity: 1
+    }],
+    discount: {
+        type: 'percent',
+        value: 10
+    },
+    bundleProducts: [ '123456789', '987654321' ]
+}, this)
+```
+> In this case, the configured discount value will be applied to all the products in the bundle. Assume for e.g., product `123456789` costs `126$` and product `987654321` costs `100$`, with a discount of 10%, 
+> the final price the customer pays will be, (126 * 0.1) + (100 * 0.1) => 113.4 + 90 => 203.4
+
+An example where NOT all the products from a bundle are added to cart.
+
+```javascript
+_targetWindow.Nosto.addBundleToCartWithDiscount({
+    items: [{
+        productId: '123456789',
+        quantity: 1
+    }],
+    discount: {
+        type: 'percent',
+        value: 10
+    },
+    bundleProducts: [ '123456789', '987654321' ]
+}, this)
+```
+
+> In this case, no discount will be offered and the customer will pay the actual price of the selected product
+
+### Shopify line item script changes
```

Comment:
Better to link to [Shopify documentation](https://help.shopify.com/en/manual/checkout-settings/script-editor) on how to upload the script to the store. That way, if the steps change, we don't need to update our documentation. You can simply say to upload the script shown below to the script editor.

Reply:
Done